database/neo4j/neoDataGeneration.js

```diff
@@ -0,0 +1,180 @@
+
+const tags = ['Good for groups', 'Desserts', 'Appetizers', 'Drinks', 'Kid friendly'];
+
+const circleColors = ['#6c8ae4', '#d86441', '#bb6acd', '#df4e96'];
+
+const jabber = new Jabber(buzzWords, 2);
+
+const allWritesFinished = [false, false];
+
+
+
+function createReviewers(NumberOfPeople) {
+  let reviewerCount = NumberOfPeople || 100;
+  const writeReviewers = fs.createWriteStream('/home/mrclean/HRR43/SDC/reviews/CSV/Reviewers.csv');
+  let ok = true;
+
+  writeReviewerToCSV();
+  function writeReviewerToCSV() {
+    do {
+      let id = reviewerCount;
+      let color = faker.random.arrayElement(circleColors);
+      let vip = Math.random() < 0.3;
+      let firstName = (Math.random() < 0.7) ? faker.name.firstName() : '';
+      let lastName;
+      if (firstName.length) {
+        lastName = (Math.random() < 0.7) ? faker.name.lastName() : '';
+      } else {
+        lastName = '';
+      }
+      reviewerCount -= 1;
+
+      if (reviewerCount === -1) {
+        writeReviewers.end();
+      } else {
+        ok = writeReviewers.write(
+          `${id},${color},${vip},${firstName},${lastName}\n`
+        );
+      }
+    } while (reviewerCount >= 0 && ok);
+
+    if (reviewerCount > 0) {
+      writeReviewers.once('drain', writeReviewerToCSV);
+    }
+  }
+};
+
+
+// re-write to be independent and select random users && random restaurants
+function createReviews(numberOfReviews){
+  const writeReviews = fs.createWriteStream('/home/mrclean/HRR43/SDC/reviews/CSV/Reviews.csv');
+  // writeReviews.write(`id, restaurantId, reviewerId, overall, food, service, ambience, dineDate, noise, recommend, filterTag, comments\n`)
+  let reviewIndex = 1;
+
+  let reviewOk = true;
+  writeReviewToCSV();
+
+  function writeReviewToCSV() {
+    do {
+      let id = reviewIndex;
+      let overall = faker.random.number({min:1, max: 5});
+      let food = faker.random.number({min:1, max: 5});
+      let service = faker.random.number({min:1, max: 5});
+      let ambience = faker.random.number({min:1, max: 5});
+      let noise = faker.random.number({min:1, max: 5});
+      let recommend = Math.random() < 0.7;
+      let comments = jabber.createParagraph(faker.random.number({min:10, max: 20}));
+
+      reviewIndex += 1;
+
+      numberOfReviews -= 1;
+
+      if (numberOfReviews === -1) {
+        writeReviews.end();
+      } else {
+        reviewOk = writeReviews.write(`${id},${overall},${food},${service},${ambience},${noise},${recommend},${comments}\n`);
+      }
+      //
+    } while (numberOfReviews >= 0 && reviewOk);
+
+    if (numberOfReviews > 0) {
+      writeReviews.once('drain', writeReviewToCSV);
+    }
+  }
+  return numberOfReviews;
+}
+
+const createRestaurants = (numberOfRestaurants) => {
+  const writeFile = fs.createWriteStream('/home/mrclean/HRR43/SDC/reviews/CSV/Restaurants.csv');
+
```

Comment:
I'm guilty of this as well, but perhaps your path would better belong in a .env file.

Reply:
oof.  good point! I'll go ahead and do that now.